OpenRA.Mods.Common/HitShapes/IHitShape.cs

```diff
@@ -0,0 +1,23 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public enum HitShape { Circle, Rectangle, Capsule }
+
+	public interface IHitShape
+	{
+		int OuterRadius { get; }
+		int DistanceFromEdge(int2 p);
```

Comment:
This should take a `WVec` and return a `WDist` to cleanly integrate with the rest of our world coordinates.